dts/arm/st/stm32f4.dtsi

```diff
@@ -135,16 +135,6 @@
 			status = "disabled";
 			label = "SPI_1";
 		};
-
-		spi2: spi@40003800 {
```

Comment:
all stm32f4 boards supported now by Zephyr have a SPI2 port. We could leave it as it is for now

Reply:
I moved SPI2 to stm32f401.dtsi, which is included directly or through other includes by every stm32f4 board that is supported by Zephyr right now. 
Following the principle from previous discussion on this PR
> dts representation should be faithful to each SoC
So I'd rather move it now, then have to do it later, when new SoCs are introduced